### PR TITLE
Add datablock group by

### LINKED ISSUE #696: Add group by for datablock
**Summary**
Add group by function to datablock.

```
group_by(block:DataBlock, columns: &[string]) -> HashMap<Vec<u8>, (Vec<u32>, Vec<DataValue>)>
```

### PR DESCRIPTION
**Pull request by BohuTANG**
## Summary

* Move `group by` function to datablocks kernel
* Change rust-toolchain from `nightly-2021-05-06` to `nightly-2021-06-01`, seems the panic is gone

## Changelog

- Improvement

## Related Issues

Fixes #696 

## Test Plan

Unit Tests
Stateless Tests

### CODE REVIEW

File: common/datablocks/src/kernels/data_block_groupby.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    /// 4, 2, 5
+    ///
+    /// 2) make indices group(for vector compute)
+    /// group_key, indices
+    /// 0, [2]
+    /// 1, [0, 3]
+    /// 2, [1, 4]
+    ///
+    /// 3) make blocks
+    pub fn group_by(block: &DataBlock, column_names: &[String]) -> Result<GroupBlocksTable> {
```
Comment (BohuTANG) [inline common/datablocks/src/kernels/data_block_groupby.rs:43]: it would be better to return vec

**Comment by BohuTANG** [comment]
> /runperf

It's unavailable, I will try to fix it.

**Review by sundy-li — APPROVED** [review]

**Comment by BohuTANG** [comment]
@sundy-li 

`/runperf` working but not show in PR's actions list, it show in:
https://github.com/datafuselabs/datafuse/actions/workflows/performance-tests-standalone.yml

**Comment by BohuTANG** [comment]
Seems it's broken.
We still need find another way to do it

**Pull request merged**